File: cpp/Reshape the Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int sum_counter = 0;
        for (vector<int> row : nums) {
            sum_counter += row.size();
        }
        if (sum_counter < r * c)
            return nums;
        else {
            vector<vector<int>> result;
            vector<int> temp;
            for (vector<int> row : nums) {
                for (int num : row) {
                    temp.push_back(num);
                    if (temp.size() == c) {
                        result.push_back(temp);
                        temp = vector<int>();
                    }
                }
            }
            return result;
        }
    }
};

File: Python/Reshape the Matrix.py
class Solution(object):
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        temp = sum(nums, [])
        if r * c > len(temp):
            return nums
        else:
            return [temp[i * c:i * c + c] for i in xrange(r)]
